app/controllers/api/base_controller.rb

```diff
@@ -38,17 +38,20 @@ def get_resource
 
     # overwrites resource_scope in FindCommon to consider nested objects
     def resource_scope(options = {})
-      options[:association] ||= controller_name
-      if nested_obj && nested_obj.respond_to?(options[:association])
-        association = nested_obj.send(options[:association])
-        if association.respond_to?(:authorized)
-          association.authorized(options[:permission], resource_class)
-        else
-          association
-        end
-      else
-        super(options)
-      end
+      child_scope = super(options)
+      child_scope.merge(parent_scope).readonly(false)
+    end
+
+    def parent_scope(options = {})
+      parent_name, parent_resource_scope = parent_resource_details()
+
+      return resource_class.where(nil) unless parent_resource_scope
+
+      association = resource_class.reflect_on_all_associations.find {|assoc| assoc.plural_name == parent_name.pluralize}
+
+      parent_sym = association.name
```

Comment:
I 'd suggest using association.name directly in line below, it's not used anywhere else and `parent_sym` does not give it better name